### ISSUE TITLE
Issue with deployed app on shinyapps.io

### ISSUE THREAD

**Issue by Tomeriko96** [vusaverse/vvdoctor#4]
When running the app on `shinyapps.io` it shows the beginning of the UI and immediately crashes and states `Disconnected from the server.`. 

The log shows the following:

It states it cannot find the `handle_file_upload` function.

``` 
2024-04-10T06:46:00.004754+00:00 shinyapps[11705693]: Container event from container-9417603: start
2024-04-10T06:46:01.002727+00:00 shinyapps[11705693]: Running on host: b1705e926075
2024-04-10T06:46:01.007455+00:00 shinyapps[11705693]: Running as user: uid=10001(shiny) gid=10001(shiny) groups=10001(shiny)
2024-04-10T06:46:01.011317+00:00 shinyapps[11705693]: Connect version: 2023.10.0
2024-04-10T06:46:01.015585+00:00 shinyapps[11705693]: LANG: C.UTF-8
2024-04-10T06:46:01.019735+00:00 shinyapps[11705693]: Working directory: /srv/connect/apps/vvdoctor
2024-04-10T06:46:01.023550+00:00 shinyapps[11705693]: Using R 4.3.1
2024-04-10T06:46:01.028846+00:00 shinyapps[11705693]: R.home(): /opt/R/4.3.1/lib/R
2024-04-10T06:46:01.032910+00:00 shinyapps[11705693]: Content will use current R environment
2024-04-10T06:46:01.036812+00:00 shinyapps[11705693]: R_LIBS: (unset)
2024-04-10T06:46:01.040682+00:00 shinyapps[11705693]: .libPaths(): /usr/lib/R, /opt/R/4.3.1/lib/R/library
2024-04-10T06:46:01.044594+00:00 shinyapps[11705693]: shiny version: 1.7.5
2024-04-10T06:46:01.048709+00:00 shinyapps[11705693]: httpuv version: 1.6.11
2024-04-10T06:46:01.054242+00:00 shinyapps[11705693]: rmarkdown version: 2.25
2024-04-10T06:46:01.058605+00:00 shinyapps[11705693]: knitr version: 1.45
2024-04-10T06:46:01.064692+00:00 shinyapps[11705693]: jsonlite version: 1.8.8
2024-04-10T06:46:01.070829+00:00 shinyapps[11705693]: RJSONIO version: (none)
2024-04-10T06:46:01.074914+00:00 shinyapps[11705693]: htmltools version: 0.5.7
2024-04-10T06:46:01.079468+00:00 shinyapps[11705693]: reticulate version: (none)
2024-04-10T06:46:01.084038+00:00 shinyapps[11705693]: Using pandoc: /opt/connect/ext/pandoc/2.16
2024-04-10T06:46:01.089608+00:00 shinyapps[11705693]: 
2024-04-10T06:46:01.093854+00:00 shinyapps[11705693]: Starting R with process ID: '32'
2024-04-10T06:46:01.098357+00:00 shinyapps[11705693]: Shiny application starting ...
2024-04-10T06:46:01.102346+00:00 shinyapps[11705693]: 
2024-04-10T06:46:01.105985+00:00 shinyapps[11705693]: Listening on http://127.0.0.1:35423
2024-04-10T06:46:05.004534+00:00 shinyapps[11705693]: Warning: Error in handle_file_upload: could not find function "handle_file_upload"
2024-04-10T06:46:05.008712+00:00 shinyapps[11705693]:   52: server [/srv/connect/apps/vvdoctor/server.R#229]
2024-04-10T06:46:05.012808+00:00 shinyapps[11705693]:   15: <Anonymous>
2024-04-10T06:46:05.016903+00:00 shinyapps[11705693]:   13: fn
2024-04-10T06:46:05.021092+00:00 shinyapps[11705693]:    8: retry
2024-04-10T06:46:05.024904+00:00 shinyapps[11705693]:    7: connect$retryingStartServer
2024-04-10T06:46:05.029170+00:00 shinyapps[11705693]:    6: eval
2024-04-10T06:46:05.033328+00:00 shinyapps[11705693]:    5: eval
2024-04-10T06:46:05.037292+00:00 shinyapps[11705693]:    4: eval
2024-04-10T06:46:05.041211+00:00 shinyapps[11705693]:    3: eval
2024-04-10T06:46:05.045302+00:00 shinyapps[11705693]:    2: eval.parent
2024-04-10T06:46:05.049499+00:00 shinyapps[11705693]:    1: local
2024-04-10T06:46:05.053102+00:00 shinyapps[11705693]: Error in handle_file_upload(input, output, session) : 
2024-04-10T06:46:05.057190+00:00 shinyapps[11705693]:   could not find function "handle_file_upload"
```

This is related to another issue, in the repo running:

`shiny::runApp` fails to run the app. Running `diagnose()` works. It seems to be an issue with how the `ui` and `server` are created in the code

**Comment by Tomeriko96** [comment]
There were a few issues, which were resolved in commit: 92c12cf98010b37a71016c289024341b7fa9dd06

- globals.R had to be global.R
- create a app.R script which sources ui.R, server.R and global.R
- server.R contained helperfunctions which had to be moved to a different R script

**Comment by Tomeriko96** [comment]
App is live https://edulytics.shinyapps.io/vvdoctor/

🥳
